parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/ParallelConsumer.java

```diff
@@ -54,6 +55,8 @@
      */
     void poll(Consumer<ConsumerRecord<K, V>> usersVoidConsumptionFunction);
 
+    void pollBatch(int batchLevel, Consumer<List<ConsumerRecord<K, V>>> usersVoidConsumptionFunction);
```

Comment:
for extensibility, this would better be in a config object, not a separate parameter.
the single message handler case may need this as well.

i would probably overload the method name (what you call it poll) for both the single and batch cases. i.e don't have pollBatch as well.

Reply:
Unfortunately due to type erasure, we can't have unlimited overloading, if the only difference is the generics type (as this erased and the compiler can't tell the difference).

How does a batch level apply to a single record consumer function? Or are you suggesting there is no single record consumer function, only a consumer which could take a list of records of size one?

Reply:
> for extensibility, batchLevel would better be in a config object, not a separate parameter.

yeah I think we have unanimous agreement on that